ledger/catchpointtracker.go

```diff
@@ -1007,11 +1009,37 @@ func (ct *catchpointTracker) accountsUpdateBalances(accountsDeltas compactAccoun
 		}
 	}
 
-	if accumulatedChanges >= trieAccumulatedChangesFlush {
-		accumulatedChanges = 0
-		_, err = ct.balancesTrie.Commit()
-		if err != nil {
-			return
+	for key, mv := range kvDeltas {
+		if mv.oldData == nil && mv.data == nil { // Came and went within the delta span
```

Comment:
I'll revisit.  At the time I wrote `accountsNewRound`, I didn't have the old data, as I had not made the change for maintaining catchpoints yet.

Reply:
Adding this:
```
	updatedKVs = make(map[string]persistedKVData, len(kvPairs))
-	for key, value := range kvPairs {
-		if value.data != nil {
-			err = writer.upsertKvPair(key, value.data)
-			updatedKVs[key] = persistedKVData{value: value.data, round: lastUpdateRound}
+	for key, mv := range kvPairs {
+		if mv.data != nil {
+			if mv.oldData != nil && bytes.Equal(mv.oldData, mv.data) {
+				continue // changed back within the delta span
+			}
+			err = writer.upsertKvPair(key, mv.data)
+			updatedKVs[key] = persistedKVData{value: mv.data, round: lastUpdateRound}
 		} else {
+			if mv.oldData == nil { // Came and went within the delta span
+				continue
+			}
 			err = writer.deleteKvPair(key)
 			updatedKVs[key] = persistedKVData{value: nil, round: lastUpdateRound}
 		}
```

Reply:
Though such an optimization would cause `au.baseKVs.write(persistedKV, key)` not to happen.  Is that ok?

We could populate `updatedKVs` even if we are not going to DB, I guess.

Reply:
Settled on doing the same thing we do for the other types.  This commit:

https://github.com/algorand/go-algorand/pull/4686/commits/65edbae901b447bcdec3935aea0f5939a771e0ac

Reply:
@jannotti Optional - I feel like the implication (quoted) is worth adding as a comment for future maintainer consideration.

> For KVs, I think this optimization would cause a commonly used, but read-only, box to fall out of the LRU periodically, since we would skip the write and not populate updatedKVs.

I'd open a PR to make the change on my end, but I'm not sure which PR contains the commit (https://github.com/algorand/go-algorand/commit/65edbae901b447bcdec3935aea0f5939a771e0ac).  If you prefer as is, feel welcomed to resolve.

Reply:
I removed it because it showed a really weird bug that I don't know how to debug yet. (it's in go e2e, causing a panic, and I get no trace because of the various jibber jabber we do to run the tests in a process that talks to real networks, etc etc)

Reply:
ok